holidayextras/deployment-helpers
# Better node module caching

#### What does this PR do? (please provide any background)

Takes a script we've got on our internal 'tripapp' project to speed up the `npm install` step on Travis, improves its options and documentation, and adds it to this repo as a shareable script that other repos can opt into.

#### What tests does this PR have?
#### How can this be tested?

[This tripapp PR](https://github.com/holidayextras/tripapplite/pull/3867) uses it, instead of a script checked into the repo.

You can test the other options by branching from it and trying repeating the build to ensure it's cached, changing the strategies or forcing updates to make sure all that behaves as expected.

I used [a repository like this](https://github.com/shackpank/output-stuff-on-install) to verify the behaviour described when version numbers / tags are updated (basically that binning the node_modules directory is the only way to guarantee it happens)

#### Screenshots / Screencast
#### What gif best describes how you feel about this work?

![ns9tpdx](https://cloud.githubusercontent.com/assets/655660/15399279/f136ee50-1dde-11e6-95e8-6ac59bf8ff0d.gif)

---


#### Reviewers

**Review 1**

**Review 2**

**Review 3**

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.
